packages/calcite-components/src/components/swatch/swatch.scss

```diff
@@ -0,0 +1,124 @@
+/**
+ * CSS Custom Properties
+ *
+ * These properties can be overridden using the component's tag as selector.
+ *
+ * @prop --calcite-swatch-corner-radius: Specifies the component's corner radius.
+ *
+ */
+
+// AUTO-GENERATED — do not modify. Changes will be overwritten.
+//
+// Internal CSS custom properties for component use only. Overwriting is not recommended.
+//
+// --calcite-internal-swatch-size
+
+:host {
+  @apply block;
+}
+
+:host([scale="s"]) {
+  .container {
+    --calcite-internal-swatch-size: 1.5rem /* 24px */;
+  }
+}
+
+:host([scale="m"]) {
+  .container {
+    --calcite-internal-swatch-size: 2rem /* 32px */;
+  }
+}
+
+:host([scale="l"]) {
+  .container {
+    --calcite-internal-swatch-size: 2.75rem /* 44px */;
+  }
+}
+
+.container {
+  @apply relative focus-base justify-center box-border  overflow-hidden;
+  font-size: var(--calcite-internal-swatch-font-size, var(--calcite-font-size));
+  block-size: var(--calcite-internal-swatch-size, auto);
+  inline-size: var(--calcite-internal-swatch-size, auto);
+  min-inline-size: var(--calcite-internal-swatch-size, auto);
+  border-radius: var(--calcite-swatch-corner-radius, 0);
+
+  &:not(.non-interactive):hover {
+    @apply cursor-pointer;
+  }
+  &:not(.non-interactive):hover {
+    box-shadow: 0 0 0 2px var(--calcite-color-border-2);
+  }
+  &.selectable {
+    @apply cursor-pointer;
+  }
+  &:not(.non-interactive):focus {
+    @apply focus-normal;
+  }
+}
+
+.swatch {
+  position: absolute;
+  @apply inline-flex overflow-hidden absolute z-10;
+  block-size: var(--calcite-internal-swatch-size, auto);
+  inline-size: var(--calcite-internal-swatch-size, auto);
+  min-inline-size: var(--calcite-internal-swatch-size, auto);
+  border-radius: var(--calcite-swatch-corner-radius, 0);
+}
+
+:host([selected]) .swatch {
+  inset: calc(var(--calcite-internal-swatch-size) / 8);
+  block-size: calc(var(--calcite-internal-swatch-size, auto) - calc(var(--calcite-internal-swatch-size) / 4));
+  inline-size: calc(var(--calcite-internal-swatch-size, auto) - calc(var(--calcite-internal-swatch-size) / 4));
+  min-inline-size: calc(var(--calcite-internal-swatch-size, auto) - calc(var(--calcite-internal-swatch-size) / 4));
+}
+
+:host([selected]) .container {
+}
+
+:host([selected]) .image-container {
+  inset: calc(var(--calcite-internal-swatch-size) / 10);
+  box-shadow:
```

Comment:
I think this box shadow needs updating. In figma at least, this is what I have: `box-shadow: 0 0 0 2px color.text.1 inset, 0 0 0 4px color.foreground.1 inset;`. What we're looking for is a final visual of 2px of black on the outside and 2px of white on the inside.